src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,53 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.ValidationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_USER_AGE = 18;
+    private static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new NullPointerException("User can't be null");
+        }
+        if (isLoginUnique(user) && isUserAdult(user) && isPasswordLengthOk(user)) {
+            storageDao.add(user);
+        }
+        return user;
```

Comment:
```suggestion
        if (isLoginUnique(user) && isUserAdult(user) && isPasswordLengthOk(user)) {
            return storageDao.add(user);
        }
        throw new RuntimeException("...some informative message");
```

we should get an exception in case user data is invalid